changelogs/unreleased/gh-6403-index-iterator-methods.md

```diff
@@ -0,0 +1,7 @@
+## bugfix/box
+
+* Fixed a bug when method `drop_while` of index iterators (`index:pairs()`,
+  `space:pairs()`) dropped an extra element (gh-6403).
+* Disabled methods `is_null` and `cycle` of index iterators
+  (`index:pairs()`, `space:pairs()`) since they worked incorrectly.
+  Now they raise an error when called.
```

Comment:
```suggestion
* Fixed a bug when the `drop_while` method of index iterators (`index:pairs()`,
  `space:pairs()`) dropped an extra element (gh-6403).
* Disabled the `is_null` and `cycle` methods of index iterators
  (`index:pairs()`, `space:pairs()`) since they worked incorrectly.
  They now raise an error when called.
```